test/views/UnderpaymentReasonSummaryViewSpec.scala

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views
+
+import base.ViewBaseSpec
+import forms.UnderpaymentReasonSummaryFormProvider
+import messages.{BaseMessages, ReasonSummary}
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.data.Form
+import play.api.mvc.Call
+import play.twirl.api.Html
+import uk.gov.hmrc.govukfrontend.views.Aliases.SummaryList
+import views.html.UnderpaymentReasonSummaryView
+
+class UnderpaymentReasonSummaryViewSpec extends ViewBaseSpec with BaseMessages {
+
+  private lazy val injectedView: UnderpaymentReasonSummaryView = app.injector.instanceOf[UnderpaymentReasonSummaryView]
+
+  val formProvider: UnderpaymentReasonSummaryFormProvider = injector.instanceOf[UnderpaymentReasonSummaryFormProvider]
+  val tempChangeAndDeleteLink: Call = Call("GET", controllers.routes.UnderpaymentReasonSummaryController.onLoad().url)
+
+  val summaryList: Option[SummaryList] = Some(SummaryList())
+
+
+  "Rendering the Reason summary page" when {
+    "no errors exist" should {
+
+      val form: Form[Boolean] = formProvider.apply()
+      lazy val view: Html = injectedView(form, tempChangeAndDeleteLink, summaryList)(fakeRequest, messages)
```

Comment:
If summaryList doesn't have any data in it, shouldn't this produce an error?

Also I think this needs a test to ensure the Summary List is rendered correctly  for the cases where there 1 one reason and 2 reasons.